File: SecondYearCS/OOP/CivitasRB/lib/SorpresaJugadorEspeculador.rb
require_relative 'sorpresa.rb'
require_relative 'JugadorEspeculador.rb'
module CivitasRB
  class SorpresaJugadorEspeculador < Sorpresa
    attr_accessor :valor
    #Constructores por defecto y con parametro asdfas
    def initialize(valor,texto)
      super(texto)
      @valor = valor
    end

    def aplicarAJugador(act, todos)
      if(self.jugadorCorrecto(act,todos))
        self.informe(act,todos)
        newJugador = JugadorEspeculador.new(todos.get(act),@valor)
        lista = todos.get(act).getPropiedades()
        lista.each do |prop|
          prop.actualizaPropietarioPorConversion(newJugador)
        end
        todos.delete_at(act)
        todos.add(act,newJugador)
      end
    end

  end
end

File: SecondYearCS/OOP/Civitas/src/civitas/SorpresaJugadorEspeculador.java
package civitas;

import java.util.ArrayList;

public class SorpresaJugadorEspeculador extends Sorpresa{

    private float valor;
    
    public SorpresaJugadorEspeculador(String texto,float valor) {
        super(texto);
        this.valor = valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador>todos){
        if(super.jugadorCorrecto(actual, todos)){
            super.informe(actual, todos);
            JugadorEspeculador newJugador = new JugadorEspeculador(todos.get(actual),valor);
            ArrayList<TituloPropiedad> lista = todos.get(actual).getPropiedades();
            for(int i=0;i<lista.size();i++){
                lista.get(i).actualizaPropietarioPorConversion(newJugador);
            }
            todos.remove(actual);
            todos.add(actual, newJugador);
           
        }
    }
    
}
